gallery/tutorials/ctf.py

```diff
@@ -0,0 +1,414 @@
+
+import aspire
+
+# Resolution to use throughout the demo.
+RESOLUTION = 512
+
+# %%
+# Visualizing the CTF
+# -------------------
+# ASPIRE can be used create and visualize example CTFs.
+
+# %%
+# Radially Symmetric CTF
+# ^^^^^^^^^^^^^^^^^^^^^^
+# The ``RadialCTFFilter`` is used to create a radially symmetric
+# ``Filter`` object for use with ASPIRE.
+# This object can also be used to calculate the transfer function as a numpy array.
+
+from aspire.image import Image
+from aspire.operators import CTFFilter, RadialCTFFilter
+
+radial_ctf_filter = RadialCTFFilter(
+    pixel_size=1,  # Angstrom
+    voltage=200,  # kV
+    defocus=10000,  # Angstrom, 10000A = 1um
+    Cs=2.26,  # Spherical aberration constant
+    alpha=0.07,  # Amplitude contrast phase in radians
+    B=0,  # Envelope decay in inverse square angstrom (default 0)
+)
+
+# The CTF filter can be visualized as an image once it is evaluated at a specific resolution.
+# More specifically the following code will return a transfer function as an array,
+# which is then plotted.
+rctf_fn = radial_ctf_filter.evaluate_grid(RESOLUTION)
+plt.imshow(rctf_fn)
+plt.colorbar()
+plt.show()
+
+# %%
+# Asymmetric CTF
+# ^^^^^^^^^^^^^^
+# For the general ``CTFFilter``, we provide defocus u and v seperately,
+# along with a defocus angle.
+
+ctf_filter = CTFFilter(
+    pixel_size=1,  # Angstrom
+    voltage=200,  # kV
+    defocus_u=15000,  # Angstrom, 10000A = 1um
+    defocus_v=10000,
+    defocus_ang=np.pi / 4,  # Radians
+    Cs=2.26,  # Spherical aberration constant
+    alpha=0.07,  # Amplitude contrast phase in radians
+    B=0,  # Envelope decay in inverse square angstrom (default 0)
+)
+
+# Again, we plot it, and note the difference from the RadialCTFFilter.
+plt.imshow(ctf_filter.evaluate_grid(RESOLUTION))
+plt.colorbar()
+plt.show()
+
+# %%
+# Phase Flipping
+# ^^^^^^^^^^^^^^
+# A common technique used with CTF corrupted data is to apply a transformation
+# based on the sign of the (estimated) CTF.
+# We can easily visuallize this in an idealized way by taking the sign of the
+# array returned by ASPIRE's ``CTFFilter.evaluate_grid``.
+
+
+ctf_sign = np.sign(radial_ctf_filter.evaluate_grid(RESOLUTION))
+plt.imshow(ctf_sign)
+plt.colorbar()
+plt.show()
+
+
+# %%
+# Applying CTF Directly to Images
+# -------------------------------
+# Various defocus values and phase flippig correction will be
+# applied directly to images using a mix of ASPIRE and Numpy.
+
+
+# %%
+# Generate example image
+# ^^^^^^^^^^^^^^^^^^^^^^
+def genExImg(L, noise_variance=0.1):
+    """
+    Generates data similar to the MATH586 lecture notes.
+
+    :param L: Resolution in pixels.
+    :param noise_variance: Variance passed to random normal noise generator.
+    :return: Image as np array.
+    """
+    # Empty Array
+    img = np.zeros((L, L))
+
+    # Make central square
+    center = L // 2
+    half_width = L // 6
+    lb, ub = center - half_width, center + half_width
+    img[lb:ub, lb:ub] = 1
```

Comment:
sure.  I updated.  Kept the floor division because the result is a little nicer (both were okay though, this one just resolves better in the idealized case).